src/Storages/RabbitMQ/StorageRabbitMQ.cpp

```diff
@@ -1053,50 +1057,20 @@ void registerStorageRabbitMQ(StorageFactory & factory)
 {
     auto creator_fn = [](const StorageFactory::Arguments & args)
     {
-        ASTs & engine_args = args.engine_args;
-        size_t args_count = engine_args.size();
-        bool has_settings = args.storage_def->settings;
 
         auto rabbitmq_settings = std::make_unique<RabbitMQSettings>();
-        if (has_settings)
-            rabbitmq_settings->loadFromQuery(*args.storage_def);
-
-        // Check arguments and settings
-        #define CHECK_RABBITMQ_STORAGE_ARGUMENT(ARG_NUM, ARG_NAME)                                           \
-            /* One of the three required arguments is not specified */                                       \
-            if (args_count < (ARG_NUM) && (ARG_NUM) <= 2 && !rabbitmq_settings->ARG_NAME.changed)            \
-            {                                                                                                \
-                throw Exception("Required parameter '" #ARG_NAME "' for storage RabbitMQ not specified",     \
-                    ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);                                           \
-            }                                                                                                \
-            if (args_count >= (ARG_NUM))                                                                     \
-            {                                                                                                \
-                if (rabbitmq_settings->ARG_NAME.changed) /* The same argument is given in two places */      \
```

Comment:
Why don't you need to check for duplicates in settings anymore? And does current code even check for them at all?

Reply:
No, I do not like this check because it is pointless and also ugly (with #define), so I deleted it.